dataplane/ebpf/src/ingress/udp.rs

```diff
@@ -95,22 +104,89 @@ pub fn handle_udp_ingress(ctx: TcContext) -> Result<i32, i64> {
         return Ok(TC_ACT_PIPE);
     }
 
-    // Calculate l3 cksum
-    // TODO(astoycos) use l3_cksum_replace instead
-    unsafe { (*ip_hdr).check = 0 };
-    let full_cksum = unsafe {
-        bpf_csum_diff(
-            mem::MaybeUninit::zeroed().assume_init(),
+    // inspired by https://github.com/torvalds/linux/blob/master/samples/bpf/tcbpf1_kern.c
+    // update dst_addr in the ip_hdr
+    // recalculate the checksums
+    unsafe {
+        let backend_ip = backend.daddr.to_be();
```

Comment:
Can you condense all of this into a function in `until.rs` that can be called in both places `ingress/udp.rs` and `ingress/tcp.rs` rather than duplicating this logic?

Reply:
now I have moved these functions to `utils.rs`.